### DISCUSSION TITLE
Symfony ApiFilter[SearchFilter]

### DISCUSSION THREAD

**Discussion by mirjalol1998bek** [symfony/symfony#53947] [Q&A]
Please help me. I've got error in ApiFilter[SearchFilter] 
Error code is :
{"@context":"\/api\/contexts\/Error","@type":"hydra:Error","hydra:title":"An error occurred","hydra:description":"Undefined constant Doctrine\\DBAL\\Types\\Types::ARRAY","trace":[{"namespace":"","short_class":"","class":"","type":"","function":"","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Doctrine\/Orm\/Filter\/SearchFilter.php","line":373,"args":[]},{"namespace":"ApiPlatform\\Doctrine\\Orm\\Filter","short_class":"SearchFilter","class":"ApiPlatform\\Doctrine\\Orm\\Filter\\SearchFilter","type":"-\u003E","function":"getType","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Doctrine\/Common\/Filter\/SearchFilterTrait.php","line":67,"args":[]},{"namespace":"ApiPlatform\\Doctrine\\Orm\\Filter","short_class":"SearchFilter","class":"ApiPlatform\\Doctrine\\Orm\\Filter\\SearchFilter","type":"-\u003E","function":"getDescription","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/OpenApi\/Factory\/OpenApiFactory.php","line":575,"args":[]},{"namespace":"ApiPlatform\\OpenApi\\Factory","short_class":"OpenApiFactory","class":"ApiPlatform\\OpenApi\\Factory\\OpenApiFactory","type":"-\u003E","function":"getFiltersParameters","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/OpenApi\/Factory\/OpenApiFactory.php","line":277,"args":[]},{"namespace":"ApiPlatform\\OpenApi\\Factory","short_class":"OpenApiFactory","class":"ApiPlatform\\OpenApi\\Factory\\OpenApiFactory","type":"-\u003E","function":"collectPaths","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/OpenApi\/Factory\/OpenApiFactory.php","line":97,"args":[]},{"namespace":"ApiPlatform\\OpenApi\\Factory","short_class":"OpenApiFactory","class":"ApiPlatform\\OpenApi\\Factory\\OpenApiFactory","type":"-\u003E","function":"__invoke","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Symfony\/Bundle\/SwaggerUi\/SwaggerUiProvider.php","line":75,"args":[]},{"namespace":"ApiPlatform\\Symfony\\Bundle\\SwaggerUi","short_class":"SwaggerUiProvider","class":"ApiPlatform\\Symfony\\Bundle\\SwaggerUi\\SwaggerUiProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Symfony\/Security\/State\/AccessCheckerProvider.php","line":53,"args":[]},{"namespace":"ApiPlatform\\Symfony\\Security\\State","short_class":"AccessCheckerProvider","class":"ApiPlatform\\Symfony\\Security\\State\\AccessCheckerProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/State\/Provider\/DeserializeProvider.php","line":47,"args":[]},{"namespace":"ApiPlatform\\State\\Provider","short_class":"DeserializeProvider","class":"ApiPlatform\\State\\Provider\\DeserializeProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Symfony\/Security\/State\/AccessCheckerProvider.php","line":53,"args":[]},{"namespace":"ApiPlatform\\Symfony\\Security\\State","short_class":"AccessCheckerProvider","class":"ApiPlatform\\Symfony\\Security\\State\\AccessCheckerProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Symfony\/Validator\/State\/QueryParameterValidateProvider.php","line":42,"args":[]},{"namespace":"ApiPlatform\\Symfony\\Validator\\State","short_class":"QueryParameterValidateProvider","class":"ApiPlatform\\Symfony\\Validator\\State\\QueryParameterValidateProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Symfony\/Validator\/State\/ValidateProvider.php","line":32,"args":[]},{"namespace":"ApiPlatform\\Symfony\\Validator\\State","short_class":"ValidateProvider","class":"ApiPlatform\\Symfony\\Validator\\State\\ValidateProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Symfony\/Security\/State\/AccessCheckerProvider.php","line":53,"args":[]},{"namespace":"ApiPlatform\\Symfony\\Security\\State","short_class":"AccessCheckerProvider","class":"ApiPlatform\\Symfony\\Security\\State\\AccessCheckerProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/State\/Provider\/ContentNegotiationProvider.php","line":56,"args":[]},{"namespace":"ApiPlatform\\State\\Provider","short_class":"ContentNegotiationProvider","class":"ApiPlatform\\State\\Provider\\ContentNegotiationProvider","type":"-\u003E","function":"provide","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/api-platform\/core\/src\/Documentation\/Action\/EntrypointAction.php","line":52,"args":[]},{"namespace":"ApiPlatform\\Documentation\\Action","short_class":"EntrypointAction","class":"ApiPlatform\\Documentation\\Action\\EntrypointAction","type":"-\u003E","function":"__invoke","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/symfony\/http-kernel\/HttpKernel.php","line":178,"args":[]},{"namespace":"Symfony\\Component\\HttpKernel","short_class":"HttpKernel","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"-\u003E","function":"handleRaw","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/symfony\/http-kernel\/HttpKernel.php","line":76,"args":[]},{"namespace":"Symfony\\Component\\HttpKernel","short_class":"HttpKernel","class":"Symfony\\Component\\HttpKernel\\HttpKernel","type":"-\u003E","function":"handle","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/symfony\/http-kernel\/Kernel.php","line":185,"args":[]},{"namespace":"Symfony\\Component\\HttpKernel","short_class":"Kernel","class":"Symfony\\Component\\HttpKernel\\Kernel","type":"-\u003E","function":"handle","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/symfony\/runtime\/Runner\/Symfony\/HttpKernelRunner.php","line":35,"args":[]},{"namespace":"Symfony\\Component\\Runtime\\Runner\\Symfony","short_class":"HttpKernelRunner","class":"Symfony\\Component\\Runtime\\Runner\\Symfony\\HttpKernelRunner","type":"-\u003E","function":"run","file":"\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/autoload_runtime.php","line":29,"args":[]},{"namespace":"","short_class":"","class":"","type":"","function":"require_once","file":"\/home\/mirjalol\/web\/symfony\/book-api\/public\/index.php","line":5,"args":[["string","\/home\/mirjalol\/web\/symfony\/book-api\/vendor\/autoload_runtime.php"]]}]}

**Comment by MatTheCat** [comment]
This issue already has been reported: https://github.com/api-platform/core/issues/6150

**Comment by mirjalol1998bek** [comment]
thank you a lot


чт, 15 февр. 2024 г. в 13:45, Mathieu ***@***.***>:
